src/storage/v2/inmemory/storage.cpp

```diff
@@ -1318,6 +1343,24 @@ void InMemoryStorage::InMemoryAccessor::Abort() {
             switch (current->action) {
               case Delta::Action::SET_PROPERTY: {
                 edge->properties.SetProperty(current->property.key, *current->property.value);
+                if (!FLAGS_storage_properties_on_edges) break;
+
+                const auto &edge_types = index_stats.property_edge_type.p2et.find(current->property.key);
+                if (edge_types != index_stats.property_edge_type.p2et.end()) {
+                  auto current_value = edge->properties.GetProperty(current->property.key);
+                  if (!current_value.IsNull()) {
+                    for (const auto &edge_type : edge_types->second) {
+                      auto *from_vertex = current->property.out_vertex;
+                      for (const auto &out_edge : from_vertex->out_edges) {
+                        if (std::get<0>(out_edge) == edge_type) {
+                          edge_property_cleanup[std::make_pair(edge_type, current->property.key)].emplace_back(
```

Comment:
With CTAD you can just use std::pair{..., ...} these day, std::make_pair should no longer be needed.